int32 cliff1
int32 cliff2
int32 cliff3
int32 cliff4
